File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int maxjump = 0;
        int n = nums.size();
        for(int i=0; i<n; i++){
            if (i>maxjump) return false;
            if ((nums[i]+i) > maxjump){
                maxjump = nums[i]+i;
            }
            if (maxjump>=n-1){
                return true;
            }

        }return false;
    }
};

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        maxjump = 0
        n = len(nums)
        for i in range (n):
            if i > maxjump:
                return False
            if (nums[i]+i)>=maxjump:
                maxjump=(nums[i]+i)
            if maxjump>=n-1:
                return True
        
        return False

        